src/polynomial.rs

```diff
@@ -164,17 +164,44 @@ pub fn poly_fft<F: FieldElement>(
     fft_interpolate_raw(points_out, points_in, n_points, scaled_roots, invert, mem)
 }
 
-pub fn poly_horner_eval<F: FieldElement>(poly: &[F], eval_at: F, len: usize) -> F {
-    let mut result = poly[len - 1];
+// Evaluate a polynomial using Horner's method.
+pub fn poly_eval<F: FieldElement>(poly: &[F], eval_at: F) -> F {
```

Comment:
This is tricky because sometimes you have a polynomial that is a `Vec<F>` and other times you have a `&[F]` or `&mut [F]`. I guess we could do something like this:
```rust
pub(crate) struct Polynomial<'a, F: FieldElement>(pub(crate) &'a [F]);
```
and construct the polynomial as needed. E.g., `Polynomial(&vec).deg()` instead of `poly_deg(&vec)`. Personally, I think the latter is cleaner. I'd prefer to keep things as they are and try to make it clear in the code and in the docs when a variable is a vector or a polynomial.

Reply:
We talked about it offline and agreed that this doesn't need to be addressed in this PR, since the representation of a polynomial predates this change.